lasy/utils/grid.py

```diff
@@ -34,23 +34,60 @@ class Grid:
         Whether the field provided uses the (complex) envelope representation, as
         used internally in lasy. If False, field is assumed to represent the
         the full (real) electric field (with fast oscillations).
+
+    is_cw : bool (optional)
+        Whether the laser pulse longitudinal profile is a continuous wave laer profile
+        or not.
+
+    is_plane_wave : bool (optional)
+        Whether the laser pulse transverse profile is a plane wave laer profile or not.
     """
 
-    def __init__(self, dim, lo, hi, npoints, n_azimuthal_modes=None, is_envelope=True):
+    def __init__(
+        self,
+        dim,
+        lo,
+        hi,
+        npoints,
+        n_azimuthal_modes=None,
+        is_envelope=True,
+        is_cw=False,
+        is_plane_wave=False,
+    ):
         # Metadata
         ndims = 2 if dim == "rt" else 3
         assert dim in ["rt", "xyt"]
         assert len(lo) == ndims
         assert len(hi) == ndims
 
-        self.lo = list(lo)
-        self.hi = list(hi)
+        lo = list(lo)
+        hi = list(hi)
+
+        if is_cw:
+            lo[-1] = -1.0
+            hi[-1] = 1.0
```

Comment:
Having a box length of 1 in SI (here 1 second) could come in handy to connect energy and power (i.e., the value of the energy in the box in J = the power in J/s).
```suggestion
            lo[-1] = -0.5
            hi[-1] = 0.5
```
Same can be done below for plane wave in transverse position.